community/primitive-collections/src/main/java/org/neo4j/concurrent/RecentK.java

```diff
@@ -30,116 +33,101 @@
  */
 public class RecentK<Type> implements Iterable<Type>
 {
+    /** Cut a number in half. */
+    public static final LongUnaryOperator COUNTER_HALFLIFE = new LongUnaryOperator()
```

Comment:
I'm not familiar with this one - is the JVM smart enough to realize it can replace that object allocation with a constant, if I put the `new LongUnaryOperator() {..}` in the method body? I'd think there'd still be obligations about each object having a unique identity hash and whatnot.